app/services/subscriptions/terminate_service.rb

```diff
@@ -197,8 +202,25 @@ def on_termination_credit_note
       @on_termination_credit_note
     end
 
-    def update_on_termination_credit_note!
-      Subscriptions::UpdateService.call!(subscription:, params: {on_termination_credit_note:})
+    def on_termination_invoice
+      return :generate if @on_termination_invoice.blank? || @on_termination_invoice.to_sym == :generate
+      return :skip if @on_termination_invoice.to_sym == :skip
+
+      # This will cause a validation error on update
+      @on_termination_invoice
+    end
```

Comment:
I found this a little surprising, but it's not important. I'd have done this when assigning 🤔 
```rb
@on_termination_invoice = on_termination_invoice.to_sym || :generate
```

Reply:
Nice point! Maybe some AI weirdness. I've updated it.